### PR TITLE
[now-cli] (Major) Upload `.env` files but ignore `.local` files

### PR DESCRIPTION
**Pull request by ijjk**
As discussed this updates to no longer ignore all `.env` files by default and only ignores local env files which matches what we default the `.gitignore` to in our default `create-next-app` [template](https://github.com/zeit/next.js/blob/20b62de8d75dd89d970457162fba20887d99e1a4/packages/create-next-app/templates/default/gitignore#L26-L30)

A separate docs PR has been made here https://github.com/zeit/docs/pull/1808 which can be landed after this PR

### CODE REVIEW

**Review by timneutkens — COMMENTED** [review]
Looks good to me, will defer to @leo and @styfle to review though 👍

File: packages/now-cli/src/util/ignored.ts [hunk 1]
```diff
@@ -14,8 +14,8 @@ export default `.hg
 .DS_Store
 .wafpicke-*
 .lock-wscript
-.env
-.env.*
+.env.local
+.env.*.local
```
Comment (styfle) [inline packages/now-cli/src/util/ignored.ts:18]: This also needs to be updated in `now-client`.

**Review by styfle — APPROVED** [review]

**Review by styfle — CHANGES_REQUESTED** [review]
*Update*: See PR #3980 which added support for preventing `.env` files from showing up in build outputs. We need to preserve this test so so I think that we need to update `@now/static-build` to add `.concat(['.env', '.env.*', 'package-lock.json', 'package.json']);`

File: packages/now-cli/src/util/ignored.ts (hunk 1 above)
Comment (ijjk) [reply packages/now-cli/src/util/ignored.ts:18]: Updated

**Review by styfle — APPROVED** [review]
Great, thanks!

**Pull request merged**